r_chapter_boostrapping_resampling.qmd

```diff
@@ -20,83 +19,212 @@ data <- iris
+  theme_classic()+
+  scale_fill_manual(values = c("#CCCCFF","#666699"),name = "", labels = c("Removed of \n the subsample", "Kept in \n the subsample"))+
+  ylab("Subsample number")+
+  xlab("Individual number")+
+  labs(title = "Representation of the subsamples in the leave-one-out \n jackknife method")
+```
+
+## Calculation of pseudo-values
 
-... 
+The next step is to calculate pseudo-values for each of the new subsamples. The formula for pseudo-values depend on the statistic of interest. In our case we want to estimate the mean, the formula will then be:
 
+$$
+v_{i} = n\overline{X} - (n-1)\overline{X}_{-i}
+$$
 
-## a - Principe
+with the following variables:
 
-## b - exemple d'implémentation
+|      Variable       | Meaning                                                                             |
+|:-------------------:|-------------------------------------------------------------------------------------|
+|       $v_{i}$       | Pseudo value of the $i^{th}$ subsample                                              |
+|         $n$         | Total number of individuals                                                         |
+|   $\overline{X}$    | Mean of the initial sample                                                          |
+| $\overline{X}_{-i}$ | Mean of the $i^{th}$ subsample, corresponding to all individuals except the $i{th}$ |
 
-test pas du tout fini
+Let's write a function which create the subsamples and calculate their pseudo values:
 
 ```{r}
-group <- c()
-sampled_data <- c()
-for (i in 1:20) { #au total, 100 individus auront été échantillonnés
-  set.seed(i)
-  sample_temp <- sample(setdiff(1:139,sampled_data),5)
-  sampled_data <- c(sampled_data,sample_temp)
-  group <- c(group,mean(data[sample_temp,"Sepal.Width"]))
-  
+pseudo_val <- function(data, theta){
+  #entry : data = the vector of data to which we want to apply the Jackknife
+  #entry : theta = function for the statistic of interest
+  #output : a vector of pseudo values for each subsample
+  n <- length(data)
+  mean_init <- theta(data)
+  pv <- rep(NA,n) #to keep in memory each pseudo value
+  for (i in 1:n) {
+    pv[i] <- n*mean_init - (n-1)*theta(data[-i])
+  }
+  return(pv)
 }
+
+```
+
+To try the function:
+
+```{r}
+(pv <- pseudo_val(data =  data_sample$Sepal.Width, 
+                  theta = mean))
+```
+
+## Test statistic
+
+The jackknife estimator of the mean will then be calculated as follow:
+
+$$
+\begin{align}
+\hat{\theta} & = \frac{\sum_{i=1}^{n}v_{i}}{n} \\
+\hat{\theta} & = \overline{v}
+\end{align}
+$$
+
+It corresponds to the mean of the pseudo values $v_i$.
+
+
+```{r}
+(mean_pv <- mean(pv))
+```
+
+The Jackknife estimator obtained is 3.15. Note that here it is the same as the initial sample mean but it is not always the case. This technique supposes that the Jackknife estimator is normally distributed and its standard error is calculated as follow:
```

Comment:
- Ne correspond pas à `mean_pv`
-  "Note that here it is the same as the initial sample mean but it is not always the case."  **A supprimer ou préciser**